edx/xblock-lti-consumer
# [BD-24] [TNL-7607] BB-3072: Move key and client management to model level.

This PR removes the LTI 1.3 private key generation and storage from the XBlock settings.
The two main benefits are:
-  The public endpoint doesn't need to load the `modulestore` to retrieve the public keyset.
-  Keys are not stored in block settings anymore (and are automatically generated when missing, making key rotations easier).

This is the first step to allow LTI launches from Studio.

**Testing instructions:**
1. Checkout the `master` branch.
2. Set up and test a LTI Launch (make a note of the `client_id`).
3. Check out this branch on your devstack.
4. Apply migrations & restart LMS.
5. Go to Django admin and check that the `lti_1p3_client_id` and `_lti_1p3_platform_private_key_id` are filled with the `client_if` from the XBlock and `_lti_1p3_platform_private_key` is filled.
5. Run a LTI launch without changing any setting.

**Reviewers:**
- [ ] @shimulch or @pcockwell
